libkbfs/favorites.go

```diff
@@ -107,6 +113,118 @@ func NewFavorites(config Config) *Favorites {
 	return newFavoritesWithChan(config, make(chan *favReq, 100))
 }
 
+type favoritesCacheForDisk struct {
+	version int
+	cache   map[Favorite]bool
+}
+type favoritesCacheEncryptedForDisk struct {
+	version        int
+	encryptedCache []byte
+}
+
+func (f *Favorites) readCacheFromDisk(ctx context.Context) error {
+	// Read the encrypted cache from disk
+	folder := f.config.StorageRoot()
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, kbfsFavoritesCacheSubfolder, user)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// decode the data from the file and ensure its version is correct
+	var decodedData favoritesCacheEncryptedForDisk
+	err = f.config.Codec().Decode(data, &decodedData)
+	if err != nil {
+		return err
+	}
+	if decodedData.version != favoritesDiskCacheStorageVersion {
+		panic("Favorites cache storage version is wrong.")
```

Comment:
I think I changed my mind about panicking here (and below).  Just return an error.

Oh, and you might as well include the unexpected version number in the error string.